### PR TITLE
Catalan Number

### PR DESCRIPTION
**Pull request by salonishah01**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Catalan Number Code in C.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1232

### CODE REVIEW

File: C/cp/catalan_number.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include <stdio.h>
```
Comment (atarax665) [inline C/cp/catalan_number.c:11]: Add description of catalan number.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Format your code according to C style guide. You may use https://www.techiedelight.com/tools/clike

File: C/cp/catalan_number.c (hunk 1 above)
Comment (salonishah01) [reply C/cp/catalan_number.c:11]: @atarax665 Added description and also formatted according to C style.

File: C/cp/catalan_number.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	int res = 0;
+	for (int i = 0; i < n; i++)
+		res += catalan(i) *catalan(n - i - 1);
+	return res;
+}
+
+int main()
+{
+	int n;
+	printf("Enter Number: ");
```
Comment (Abhijit2505) [inline C/cp/catalan_number.c:29]: instead of using variable names like n,a,p etc, please use better names like `num/number` etc.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Some small changes requested

File: C/cp/catalan_number.c (hunk 2 above)
Comment (salonishah01) [reply C/cp/catalan_number.c:29]: @Abhijit2505 Changed the variable name.

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Review by iamrajiv — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1:

**Review by Abhijit2505 — APPROVED** [review]
Perfect now

**Pull request merged**